### PR TITLE
Avoid modifying passed-in tls.Config

### LINKED ISSUE #776: credentials.NewTLS modifies passed-in tls.Config
This was introduced in https://github.com/grpc/grpc-go/commit/78e558b. Unfortunately, the solution requires cloning the incoming tls.Config, for which no exported facility exists. See https://github.com/golang/go/issues/15771 and https://github.com/golang/go/issues/16492.

### PR DESCRIPTION
**Pull request by menghanl**
fixes #776

### CODE REVIEW

**Comment by menghanl** [comment]
This change reverts 78e558b

**Comment by tamird** [comment]
No, this is not the correct fix. This copies a mutex.

**Comment by tamird** [comment]
LGTM

**Comment by iamqizhao** [comment]
LGTM, will merge once travis is green.

File: credentials/credentials.go [hunk 1]
```diff
@@ -187,9 +187,45 @@ func (c *tlsCreds) ServerHandshake(rawConn net.Conn) (net.Conn, AuthInfo, error)
...
+		InsecureSkipVerify:          cfg.InsecureSkipVerify,
+		CipherSuites:                cfg.CipherSuites,
+		PreferServerCipherSuites:    cfg.PreferServerCipherSuites,
+		SessionTicketsDisabled:      cfg.SessionTicketsDisabled,
+		SessionTicketKey:            cfg.SessionTicketKey,
+		ClientSessionCache:          cfg.ClientSessionCache,
+		MinVersion:                  cfg.MinVersion,
+		MaxVersion:                  cfg.MaxVersion,
+		CurvePreferences:            cfg.CurvePreferences,
+		DynamicRecordSizingDisabled: cfg.DynamicRecordSizingDisabled,
```
Comment (tamird) [inline credentials/credentials.go]: these last two fields don't exist in Go 1.6 or 1.5

**Comment by tamird** [comment]
This is green; let's get it merged plz.

**Pull request merged**